app/controllers/registration/ManageRegistrationController.scala

```diff
@@ -48,242 +43,171 @@ class ManageRegistrationController @Inject()(
+      cachingService.cacheBikRemoved(RegistrationItem(uriInformation.iabdValueURLDeMapper(iabdType), false, true))
       val resultFuture = Future.successful(
         Ok(
           confirmUpdateNextTaxYearView(
-            formMappings.objSelectedForm.fill(form.get),
+            registrationList,
+            Some(bikToRemove),
             additive = false,
             controllersReferenceData.YEAR_RANGE,
-            empRef = request.empRef)))
+            empRef = request.empRef
+          )))
       controllersReferenceData.responseErrorHandler(resultFuture)
     }
 
-  def removeNextYearRegisteredBenefitTypes: Action[AnyContent] = (authenticate andThen noSessionCheck).async {
-    implicit request =>
-      val persistentBiks: List[Bik] =
-        controllersReferenceData.generateListOfBiksBasedOnForm(ControllersReferenceDataCodes.BIK_REMOVE_STATUS)
-      val registeredFuture =
-        updateBiksFutureAction(controllersReferenceData.YEAR_RANGE.cy, persistentBiks, additive = false)
-      controllersReferenceData.responseErrorHandler(registeredFuture)
-  }
-
-  def generateConfirmationScreenView(
-    year: Int,
-    cachingSuffix: String,
-    generateViewBasedOnFormItems: Form[RegistrationList] => HtmlFormat.Appendable,
-    viewToRedirect: Form[RegistrationList] => Result)(
-    implicit hc: HeaderCarrier,
-    request: Request[AnyContent]): Future[Result] =
-    formMappings.objSelectedForm.bindFromRequest.fold(
-      formWithErrors => Future.successful(viewToRedirect(formWithErrors)),
-      values => {
-
-        val items: List[RegistrationItem] = values.active.filter(x => x.active)
-        Future.successful(
-          Ok(generateViewBasedOnFormItems(formMappings.objSelectedForm.fill(RegistrationList(None, items, None)))))
-
+  def removeNextYearRegisteredBenefitTypes(iabdType: String): Action[AnyContent] =
+    (authenticate andThen noSessionCheck).async { implicit request =>
+      val registeredFuture = cachingService.fetchPbikSession().flatMap { session =>
+        val bikToRemove = Bik(session.get.bikRemoved.get.id, ControllersReferenceDataCodes.BIK_REMOVE_STATUS)
+        updateBiksFutureAction(controllersReferenceData.YEAR_RANGE.cy, List(bikToRemove), additive = false)
       }
-    )
+      controllersReferenceData.responseErrorHandler(registeredFuture)
+    }
 
-  def updateRegisteredBenefitTypes: Action[AnyContent] = (authenticate andThen noSessionCheck).async {
+  def updateCurrentYearRegisteredBenefitTypes: Action[AnyContent] = (authenticate andThen noSessionCheck).async {
     implicit request =>
-      val resultFuture = {
-
-        val persistentBiks: List[Bik] =
-          controllersReferenceData.generateListOfBiksBasedOnForm(ControllersReferenceDataCodes.BIK_ADD_STATUS)
+      val actionFuture = cachingService.fetchPbikSession().flatMap { session =>
+        val persistentBiks = session.get.registrations.get.active
+          .filter(x => x.active)
```

Comment:
can you rename single letter variables to something more meaningful

Reply:
Yep good spot